tests/ui/zero_sized_map_values.rs

```diff
@@ -0,0 +1,69 @@
+#![warn(clippy::zero_sized_map_values)]
```

Comment:
Added a test case with BTreeMap. Should I add additional tests for other cases too or do we trust that BTreeMap and HashMap are treated equally?

Reply:
I think it would be preferable to check all the cases for both containers.

A simple way to go would be to copy the current test file and change the `HashMap`s to `BTreeMap`s. We can name the files `zero_sized_hashmap_values.rs` and `zero_sized_btreemap_values.rs` for example.